### PR TITLE
Fix Document vs Description terminology (3.1.1, port of #4100)

### PR DESCRIPTION
**Pull request by handrews**
I think this is correct (and includes the last remaining "suggestion" about the Server Object `url` field wording), but please review carefully.  I still need to re-review it myself, as resolving conflicts and adjusting for 3.1.1 was a bit complex in spots.

Note that #4130 does not need to be ported as it is already handled by having the two distinct "Relative References in..." (URI vs URL) sections.  Although those sections assume the split documented in #4130 that we said was ambiguous in 3.0.

### CODE REVIEW

**Review by ralfhandl — CHANGES_REQUESTED** [review]
The same three nits as in #4100

File: versions/3.1.1.md
[the remark this answers is not in the document]
Comment (handrews) [reply versions/3.1.1.md:1]: I'm guessing this is the Server Object bit- let's see what wording we settle on in 3.0.4 before updating here.

[the remark this answers is not in the document]
Comment (handrews) [reply versions/3.1.1.md:1]: @ralfhandl I've updated this with the text @mikekistler accepted in the 3.0.4 PR, so I'm going to mark this resolved.

**Review by ralfhandl — APPROVED** [review]

**Review by mikekistler — APPROVED** [review]
Looks good! 👍

**Pull request merged**